tests/summarywriter_test.cpp

```diff
@@ -107,6 +114,56 @@ TEST_CASE_METHOD(SummaryWriterTestsFixture, "Writing text summary to file",
       " Project developed during GSoc 2020 ", *f1);
 }
 
+/**
+ * Test text summary.
+ */
+TEST_CASE_METHOD(SummaryWriterTestsFixture, "Writing summary using callback to file",
```

Comment:
Let's add these tests in a separate file named `callback_test.cpp`. This file will get restructured quite a lot in future. So it's better to keep callback tests separately.

Reply:
Ok